bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -671,9 +698,8 @@ public boolean equals(Object o, Set<ValuePair> visitedValues) {
         if (o instanceof XmlSequence rhsXMLSequence) {
             return isXMLSequenceChildrenEqual(this.getChildrenList(), rhsXMLSequence.getChildrenList());
         }
-        if (o instanceof XmlItem) {
-            return this.getChildrenList().size() == 1 &&
-                    isEqual(this.getChildrenList().get(0), o);
+        if (o instanceof XmlItem || (o instanceof XmlText && this.isSingleton())) {
+            return isEqual(this.getChildrenList().get(0), o);
```

Comment:
This change isn't correct? Produces incorrect results now for

```ballerina
xml x1 = xml:concat(xml `<foo/>`, xml `<bar/>`);
xml:Element x2 = xml `<foo/>`;
boolean equal = x1 == x2;
```

Reply:
Fixed with cfc81cfad46986467573c70a6c7a316ea085d2ae